autofill/autofill-impl/src/test/java/com/duckduckgo/autofill/jsbridge/response/AutofillJsonResponseWriterTest.kt

```diff
@@ -0,0 +1,137 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.jsbridge.response
+
+import com.duckduckgo.autofill.domain.javascript.JavascriptCredentials
+import com.squareup.moshi.Moshi
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class AutofillJsonResponseWriterTest {
+    private val testee = AutofillJsonResponseWriter(Moshi.Builder().build())
+
+    @Test
+    fun whenGenerateResponseGetAutofillDataTheReturnAutofillDataJson() {
+        val expectedJson = "{\n" +
+            "  \"success\": {\n" +
+            "    \"action\": \"fill\",\n" +
+            "    \"credentials\": {\n" +
+            "      \"password\": \"password\",\n" +
+            "      \"username\": \"test\"\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"type\": \"getAutofillDataResponse\"\n" +
+            "}"
+        assertEquals(
+            expectedJson,
+            testee.generateResponseGetAutofillData(
+                JavascriptCredentials(
+                    username = "test",
+                    password = "password"
+                )
+            )
+        )
+    }
+
+    @Test
+    fun whenGenerateEmptyResponseGetAutofillDataThenReturnEmptyResponseJson() {
+        val expectedJson = "{\n" +
+            "  \"success\": {\n" +
+            "    \"action\": \"non\"\n" +
+            "  },\n" +
+            "  \"type\": \"getAutofillDataResponse\"\n" +
+            "}"
+        assertEquals(
+            expectedJson,
+            testee.generateEmptyResponseGetAutofillData()
+        )
+    }
+
+    @Test
+    fun whenGenerateResponseGetAvailableInputTypesThenReturnAvailableInputTypesJson() {
+        val expectedJson = "{\n" +
+            "  \"credentials\": false,\n" +
+            "  \"email\": true\n" +
+            "}"
+        assertEquals(
+            expectedJson,
+            testee.generateResponseGetAvailableInputTypes(
+                credentialsAvailable = false,
+                emailAvailable = true
+            )
+        )
+    }
+
+    @Test
+    fun generateContentScope() {
+        val expectedJson = "contentScope = {\n" +
+            "  \"features\": {\n" +
+            "    \"autofill\": {\n" +
+            "      \"state\": \"enabled\",\n" +
+            "      \"exceptions\": []\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"unprotectedTemporary\": []\n" +
+            "};"
+        assertEquals(
+            expectedJson,
+            testee.generateContentScope()
+        )
+    }
+
+    @Test
+    fun generateUserUnprotectedDomains() {
+        val expectedJson = "userUnprotectedDomains = [];"
+        assertEquals(
+            expectedJson,
+            testee.generateUserUnprotectedDomains()
+        )
+    }
+
+    @Test
+    fun generateUserPreferences() {
+        val expectedJson = "userPreferences = {\n" +
```

Comment:
It makes me a little anxious to do a literal string comparison for this, in that it might be unstable if the order of properties change, or the number of spaces in indentation changes etc...

But I think its addition as is is still an improvement of not having these tests.